### ISSUE TITLE
Inconsistency between spec and APG for aria-controls required on combobox

### ISSUE THREAD

**Issue by carmacleod** [w3c/aria-practices#1557]
The ARIA spec says `aria-controls` is required on [combobox](https://w3c.github.io/aria/#combobox).

The APG Roles, States and Properties section of the [combobox pattern](https://w3c.github.io/aria-practices/#wai-aria-roles-states-and-properties-6) says:
> The combobox element has `aria-controls` set to a value that refers to the element that serves as the popup. Note that `aria-controls` only needs to be set when the popup is visible. However, it is valid to reference an element that is not visible.

The 2nd sentence is problematic. It seems to be implying that `aria-controls` is not always required, i.e. authors can leave it off when the popup is not visible.

This may lead to confusion, with authors thinking, "Well, is it required or not?".
It also complicates processing for validator tools, making it less likely that tooling will catch incorrectly marked up comboboxes.

Consider rewriting the above point in the APG to say:
> The combobox element has `aria-controls` set to a value that refers to the element that serves as the popup. Note that it is valid for `aria-controls` to reference an element that is not visible.

**Comment by JAWS-test** [comment]
In my opinion, the requirement that aria-controls must be present in combobox, even if there is no content to which aria-controls refers, is absurd. This forces the developers to insert e.g. an empty, invisible listbox into the source code even when a combobox is closed. However, empty listboxes are not allowed according to the ARIA specification ([Required Owned Elements](https://www.w3.org/TR/wai-aria-1.2/#mustContain)). 
That's why I'm more in favor of aria-controls only being mandatory when the combobox is open. When closed, aria-controls can point to an invisible listbox (grid, dialog, tree), but it does not have to be

**Comment by JAWS-test** [comment]
See https://github.com/w3c/aria/issues/1334, so this issue can be closed

**Comment by carmacleod** [comment]
Agreed that the fix for https://github.com/w3c/aria/issues/1334 (in PR https://github.com/w3c/aria/pull/1335) is a better fix.
Closing.
